packages/sign_in_with_apple/test/sign_in_with_apple_button/sign_in_with_apple_button_test.dart

```diff
@@ -32,68 +34,176 @@ Future<void> main() async {
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('goldens/black_button.png'),
+      );
+    });
+
+    testWidgets('White', (tester) async {
+      await tester.pumpWidget(
+        TestSetup(
+          backgroundColor: Colors.black,
+          child: SignInWithAppleButton(
+            onPressed: () {},
+            style: SignInWithAppleButtonStyle.white,
+          ),
+        ),
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('goldens/white_button.png'),
+      );
+    });
+
+    testWidgets('White Outlined', (tester) async {
+      await tester.pumpWidget(
+        TestSetup(
+          child: SignInWithAppleButton(
+            onPressed: () {},
+            style: SignInWithAppleButtonStyle.whiteOutlined,
+          ),
+        ),
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('goldens/white_outlined_button.png'),
+      );
+    });
+  });
+
+  group('Icon Alignment', () {
+    testWidgets('Left aligned icon', (tester) async {
+      await tester.pumpWidget(
+        TestSetup(
+          child: SignInWithAppleButton(
+            onPressed: () {},
+            iconAlignment: IconAlignment.left,
+          ),
+        ),
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('goldens/left_aligned_icon.png'),
+      );
+    });
+
+    testWidgets('Center aligned Icon', (tester) async {
+      await tester.pumpWidget(
+        TestSetup(
+          child: SignInWithAppleButton(
+            onPressed: () {},
+            iconAlignment: IconAlignment.center,
+          ),
+        ),
+      );
+
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('goldens/center_aligned_icon.png'),
+      );
+    });
+  });
+
+  testWidgets('Allows to customize the border radius of the button',
+      (tester) async {
     await tester.pumpWidget(
       TestSetup(
-        child: SignInWithAppleButton(onPressed: () {}),
+        child: SignInWithAppleButton(
+          onPressed: () {},
+          borderRadius: BorderRadius.all(
+            Radius.circular(22.0),
+          ),
+        ),
       ),
     );
 
     await expectLater(
       find.byType(CupertinoApp),
-      matchesGoldenFile('goldens/default_style.png'),
+      matchesGoldenFile('goldens/custom_border_radius.png'),
     );
   });
 
-  testWidgets('Left aligned icon', (tester) async {
+  testWidgets('Allows to customize the height of the button', (tester) async {
     await tester.pumpWidget(
       TestSetup(
         child: SignInWithAppleButton(
           onPressed: () {},
-          iconAlignment: IconAlignment.left,
+          height: 60,
         ),
       ),
     );
 
     await expectLater(
       find.byType(CupertinoApp),
-      matchesGoldenFile('goldens/left_aligned_icon.png'),
+      matchesGoldenFile('goldens/custom_height.png'),
```

Comment:
I think for the custom height, the spacing between the logo and the text is too narrow.